protocols/relay/src/v2/protocol/inbound_hop.rs

```diff
@@ -136,40 +145,36 @@ pub struct ReservationReq {
 
 impl ReservationReq {
     pub async fn accept(self, addrs: Vec<Multiaddr>) -> Result<(), UpgradeError> {
-        let msg = HopMessage {
-            r#type: hop_message::Type::Status.into(),
-            peer: None,
-            reservation: Some(Reservation {
-                addrs: addrs.into_iter().map(|a| a.to_vec()).collect(),
-                expire: (SystemTime::now() + self.reservation_duration)
-                    .duration_since(SystemTime::UNIX_EPOCH)
-                    .unwrap()
-                    .as_secs(),
-                voucher: None,
-            }),
-            limit: Some(Limit {
-                duration: Some(
-                    self.max_circuit_duration
-                        .as_secs()
-                        .try_into()
-                        .expect("`max_circuit_duration` not to exceed `u32::MAX`."),
-                ),
-                data: Some(self.max_circuit_bytes),
-            }),
-            status: Some(Status::Ok.into()),
-        };
+        let mut msg = HopMessage::new();
+        msg.set_type(hop_message::Type::STATUS);
+        msg.set_status(Status::OK);
```

Comment:
Can we retain the order here please (`status` being set last)? Makes the diff easier to read :)